### ISSUE TITLE
Error: 'android.apk' is not supported

### ISSUE THREAD

**Issue by Ademar-SJ** [wix/Detox#2675]
After a hard configuration to work with React Native and the emulator MemuPlay (which connect just with adb, i can't get the so the avdName, so thats the only way i find to locate him) my settings look like this:

```json
{
  "testRunner": "jest",
  "runnerConfig": "e2e/config.json",
  "configurations": {
    "debug": {
      "binaryPath": "C:/Users/adema/Desktop/myMoney/android/app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk",
        "build": "cd android & gradlew build assembleDebug assembleAndroidTest -DtestBuildType=debug",
      "type": "android.apk",
        "devices": {
          "emulator": {
            "type": "android.attached",
            "device": {
              "adbName": "device"
            }
          }
        }
      }
  }
}
```

the build worked, however the "detox test" doesn't work, i tried change the binaryPath to the local i used before without detox (only debug/apk-debug.apk and not de AndroidTest app) but the error persist:

```
detox[12348] INFO:  [test.js] DETOX_FORCE_ADB_INSTALL=false DETOX_READ_ONLY_EMU=false DETOX_REPORT_SPECS=true DETOX_START_TIMESTAMP=1614902798530 DETOX_USE_CUSTOM_LOGGER=true jest --config e2e/config.json --testNamePattern "^((?!:ios:).)*$" --maxWorkers 1 e2e
detox[10628] INFO:  [DetoxServer.js] server listening on localhost:62082...
detox[10628] ERROR: Error: 'android.apk' is not supported
detox[10628] INFO:  Example is assigned to undefined
detox[10628] INFO:  Example: should have welcome screen
detox[10628] INFO:  Example: should have welcome screen [SKIPPED]

detox[12348] ERROR: [cli.js] Error: Command failed: jest --config e2e/config.json --testNamePattern "^((?!:ios:).)*$" --maxWorkers 1 e2e
```

someone can help please,

**Comment by noomorph** [comment]
See #2676, @Ademar-SJ

**Comment by noomorph** [comment]
And please read the docs carefully: https://github.com/wix/Detox/blob/master/docs/APIRef.Configuration.md

This is approximately how config would look for a sole Android app:

```js
module.exports = {
  "testRunner": "jest",
  "runnerConfig": 'e2e/config.json',
  "apps": {
    "android.release": {
      "type": "android.apk",
      "binaryPath": "../demo-react-native/android/app/build/outputs/apk/release/app-release.apk",
    }
  },
  "devices": {
    "emulator": {
      "type": "android.emulator",
      "device": {
        "avdName": "Pixel_API_28"
      }
    }
  },
  "configurations": {
    "android.emu.release": {
      "device": "emulator",
      "app": "android.release",
    }
  }
};
```

**Comment by noomorph** [comment]
The bad error message is fixed in [18.7.1](https://www.npmjs.com/package/detox/v/18.7.1).
